core/constant_inc.php

```diff
@@ -591,8 +592,11 @@
 define( 'DB_FIELD_SIZE_PASSWORD', 64 );
 define( 'DB_FIELD_SIZE_API_TOKEN_NAME', 128 );
 
-# Maximum size for the user's password when storing it as a hash
-define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 1024 );
+# Maximum size for the user's password when storing it as a hash.
+# The CRYPT_BLOWFISH algorithm used by password_hash() will silently truncate
+# passwords to 72 characters, so there is no point in allowing longer passwords.
+# @see http://php.net/function.crypt
+define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 72 );
```

Comment:
We really should only enforce limit on plain password if it doesn't fit into the db schema.  We should also retire plain password to get out of this business and only offer rolling it over to a secure algorithm.